sunpy/map/mapsequence.py

```diff
@@ -68,13 +76,18 @@ def __init__(self, *args, sortby='date', derotate=False, **kwargs):
 
         # Optionally sort data
         if sortby is not None:
-            if sortby == 'date':
-                self.maps.sort(key=self._sort_by_date())
-            else:
-                raise ValueError("Only sort by date is supported")
+            if sortby not in self._sort_methods:
+                raise ValueError(f"sortby must be one of the following: {self._sort_methods.keys()}")
```

Comment:
```suggestion
                raise ValueError(f"sortby must be one of the following: {list(self._sort_methods.keys())}")
```
Otherwise I think the output looks like:

>>> f'{abc.keys()}'
"dict_keys(['a', 'b'])"